### ISSUE TITLE
Multipass not recognizing Windows 11 Professional

### ISSUE THREAD

**Issue by ricab** [canonical/multipass#3250]
Hello, I have the same issue running multipass on Windows 11 Professional. 

I just did a Quick Create on Hyper-V and an Ubuntu VM was installed without any problem, also Hyper-V was enable since I installed Windows, so it was not a recent enable without reboot problem.

```
Edition	Windows 11 Pro
Version	22H2
Installed on	‎9/‎22/‎2022
OS build	22621.2361
Experience	Windows Feature Experience Pack 1000.22674.1000.0
```

```
❯ multipass networks
networks failed: Multipass support for Hyper-V requires Windows 10 or newer

❯ multipass version
multipass   1.12.2+win
multipassd  1.12.2+win
```

[edit]
Uninstalled multipass 1.12.2, disabled Hyper-V, reboot machine, enabled Hyper-V + reboot, installed multipass 1.11.1 + reboot. Same error message.
After installation, looking at the Event Viewer, there are two warnings messages. I will try to find where the installation logs are located because the detailed message during installation seems to be some errors in there.

`[daemon] Hypervisor health check failed: Multipass support for Hyper-V requires Windows 10 or newer`
`[Qt] QIODevice::write (QFile, "C:\ProgramData\Multipass\cache\vault\multipassd-image-records.json"): device not open`

[edit2]
Same issue about port 50051 https://github.com/canonical/multipass/issues/573

_Originally posted by @ltsuda in https://github.com/canonical/multipass/issues/3090#issuecomment-1751201379_

**Comment by ricab** [comment]
Hi @ltsuda, sorry you're having trouble getting it to work. Could you please run the following command in PowerShell and let us know the output?

```
Get-ItemProperty -Path 'HKLM:\\Software\\Microsoft\\Windows NT\\CurrentVersion' | Select-Object -ExpandProperty CurrentMajorVersionNumber
```

There might be some locale issue that we're not handling properly or something...

**Comment by ltsuda** [comment]
No problem,

```
❯ Get-ItemProperty -Path 'HKLM:\\Software\\Microsoft\\Windows NT\\CurrentVersion' | Select-Object -ExpandProperty CurrentMajorVersionNumber
10
```

![image](https://github.com/canonical/multipass/assets/3965277/900bc2ef-d76b-4f13-9cf0-4b8b4938009f)

**Comment by ricab** [comment]
Hmm, puzzling. Weird that it says 10 when it is 11. Anyway, 10 should work and there doesn't appear to be any problem with that string.

Would you mind posting the output of `multipass networks -vvvv`, please?

**Comment by ltsuda** [comment]
```
❯ multipass networks -vvvv
[2023-10-09T08:52:42.657] [trace] [Hyper-V Health Check] PowerShell arguments: -NoProfile, -NoExit, -Command, -
[2023-10-09T08:52:42.657] [trace] [Hyper-V Health Check] PowerShell working dir ''
[2023-10-09T08:52:42.657] [trace] [Hyper-V Health Check] PowerShell program 'powershell.exe'
[2023-10-09T08:52:42.657] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Starting
[2023-10-09T08:52:42.670] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Running
[2023-10-09T08:52:42.670] [debug] [powershell.exe] [21820] started: powershell.exe -NoProfile -NoExit -Command -
[2023-10-09T08:52:42.670] [debug] [Hyper-V Health Check] [21820] Cmdlet: 'Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion' | Select-Object -ExpandProperty CurrentMajorVersionNumber'
[2023-10-09T08:52:42.829] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:52:43.049] [trace] [Hyper-V Health Check] Cannot load PSReadline module.  Console is running without PSReadline.
10
[2023-10-09T08:52:43.049] [trace] [Hyper-V Health Check] [21820] Output: Cannot load PSReadline module.  Console is running without PSReadline.
10
[2023-10-09T08:52:43.049] [debug] [Hyper-V Health Check] [21820] Cmdlet exit status is 'true'
[2023-10-09T08:52:43.062] [trace] [Hyper-V Health Check] [21820] PowerShell state changed to NotRunning
[2023-10-09T08:52:43.062] [debug] [Hyper-V Health Check] [21820] PowerShell finished successfully
networks failed: Multipass support for Hyper-V requires Windows 10 or newer
```

```
❯ multipass launch --name master -m 6G -c 3 --disk 50G -vvvvv
[2023-10-09T08:54:58.713] [trace] [url downloader] Found https://codeload.github.com/canonical/multipass-blueprints/zip/refs/heads/main in cache: true
[2023-10-09T08:54:58.715] [debug] [blueprint provider] Loading "anbox-cloud-appliance" v1
[2023-10-09T08:54:58.716] [debug] [blueprint provider] Loading "charm-dev" v1
[2023-10-09T08:54:58.716] [debug] [blueprint provider] Loading "docker" v1
[2023-10-09T08:54:58.717] [debug] [blueprint provider] Loading "jellyfin" v1
[2023-10-09T08:54:58.717] [debug] [blueprint provider] Loading "minikube" v1
[2023-10-09T08:54:58.718] [debug] [blueprint provider] Loading "ros-noetic" v1
[2023-10-09T08:54:58.718] [debug] [blueprint provider] Loading "ros2-humble" v1
[2023-10-09T08:54:58.949] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:54:59.196] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/16/stable/current/SHA256SUMS in cache: true
[2023-10-09T08:54:59.661] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/18/stable/current/SHA256SUMS in cache: true
[2023-10-09T08:54:59.967] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:00.267] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/20/stable/current/SHA256SUMS in cache: true
[2023-10-09T08:55:00.907] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/22/stable/current/SHA256SUMS in cache: true
[2023-10-09T08:55:00.960] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:01.800] [trace] [url downloader] Found https://cloud-images.ubuntu.com/releases/streams/v1/index.json in cache: false
[2023-10-09T08:55:01.958] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:02.972] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:03.972] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:04.097] [trace] [url downloader] Found https://cloud-images.ubuntu.com/releases/streams/v1/com.ubuntu.cloud:released:download.json in cache: false
[2023-10-09T08:55:04.885] [trace] [url downloader] Found https://cloud-images.ubuntu.com/buildd/daily/streams/v1/index.json in cache: false
[2023-10-09T08:55:04.971] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:05.968] [trace] [url downloader] Found https://cloud-images.ubuntu.com/buildd/daily/streams/v1/com.ubuntu.cloud:daily:download.json in cache: false
[2023-10-09T08:55:05.968] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-09T08:55:05.978] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/appliances/streams/v1/index.json in cache: true
[2023-10-09T08:55:05.980] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/appliances/streams/v1/com.ubuntu.appliances:released:download.json in cache: true
[2023-10-09T08:55:05.982] [trace] [Hyper-V Health Check] PowerShell arguments: -NoProfile, -NoExit, -Command, -
[2023-10-09T08:55:05.982] [trace] [Hyper-V Health Check] PowerShell working dir ''
[2023-10-09T08:55:05.982] [trace] [Hyper-V Health Check] PowerShell program 'powershell.exe'
[2023-10-09T08:55:05.982] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Starting
[2023-10-09T08:55:05.988] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Running
[2023-10-09T08:55:05.989] [debug] [powershell.exe] [8504] started: powershell.exe -NoProfile -NoExit -Command -
[2023-10-09T08:55:05.989] [debug] [Hyper-V Health Check] [8504] Cmdlet: 'Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion' | Select-Object -ExpandProperty CurrentMajorVersionNumber'
[2023-10-09T08:55:06.360] [trace] [Hyper-V Health Check] Cannot load PSReadline module.  Console is running without PSReadline.
10
[2023-10-09T08:55:06.360] [trace] [Hyper-V Health Check] [8504] Output: Cannot load PSReadline module.  Console is running without PSReadline.
10
[2023-10-09T08:55:06.360] [debug] [Hyper-V Health Check] [8504] Cmdlet exit status is 'true'
[2023-10-09T08:55:06.372] [trace] [Hyper-V Health Check] [8504] PowerShell state changed to NotRunning
[2023-10-09T08:55:06.372] [debug] [Hyper-V Health Check] [8504] PowerShell finished successfully
launch failed: Multipass support for Hyper-V requires Windows 10 or newer
leona ~  7.905s
❯
```

**Comment by ricab** [comment]
Hi @ltsuda, OK this is basically the same problem as https://github.com/canonical/multipass/issues/1632. Not caused by screen reader in your case, but the underlying issue is the same: Multipass expects a "10" when running that PS command and gets that warning text about PSReadline instead. Until we get around to fixing this, here are a things to try to work around, based on what worked for others:

  - [Install PSReadline](https://github.com/canonical/multipass/issues/1341#issuecomment-726036847)
  - [Change the execution policy](https://github.com/canonical/multipass/issues/1341#issuecomment-726062102)
  - [Downgrade PSReadLine](https://github.com/canonical/multipass/issues/2140#issuecomment-864952893)
  - [Upgrade PSReadLine](https://github.com/canonical/multipass/issues/1077#issuecomment-534487167)
  
Can you give those a try and let us know how it goes?

**Comment by ltsuda** [comment]
HI @ricab, thank for all the information.

I will give it a try one by one but only after my work hours, then I return here with the results.

Appreciate the help, thank you.

**Comment by ltsuda** [comment]
Hello, I'm back.

The problem is the ExecutionPolicy that was `Undefined` for `LocalMachine`.

Configuration that raises error:
```
❯ Get-ExecutionPolicy -List

        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser    RemoteSigned
 LocalMachine       Undefined
```

Configuration that fix:
I was able to run the networks command and launch a instance without issues.

```
❯ Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope LocalMachine
❯ Get-ExecutionPolicy -List

        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser    RemoteSigned
 LocalMachine    RemoteSigned
```

```
❯ multipass networks -vvvv
[2023-10-10T14:54:13.780] [trace] [Hyper-V Health Check] PowerShell arguments: -NoProfile, -NoExit, -Command, -
[2023-10-10T14:54:13.780] [trace] [Hyper-V Health Check] PowerShell working dir ''
[2023-10-10T14:54:13.780] [trace] [Hyper-V Health Check] PowerShell program 'powershell.exe'
[2023-10-10T14:54:13.780] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Starting
[2023-10-10T14:54:13.792] [trace] [Hyper-V Health Check] [0] PowerShell state changed to Running
[2023-10-10T14:54:13.793] [debug] [powershell.exe] [16816] started: powershell.exe -NoProfile -NoExit -Command -
[2023-10-10T14:54:13.793] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion' | Select-Object -ExpandProperty CurrentMajorVersionNumber'
[2023-10-10T14:54:14.151] [trace] [Hyper-V Health Check] 10
[2023-10-10T14:54:14.151] [trace] [Hyper-V Health Check] [16816] Output: 10
[2023-10-10T14:54:14.151] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.151] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion' | Select-Object -ExpandProperty ReleaseId'
[2023-10-10T14:54:14.168] [trace] [Hyper-V Health Check] 2009
[2023-10-10T14:54:14.168] [trace] [Hyper-V Health Check] [16816] Output: 2009
[2023-10-10T14:54:14.168] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.168] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-CimInstance Win32_ComputerSystem | Select-Object -ExpandProperty HypervisorPresent'
[2023-10-10T14:54:14.261] [trace] [Hyper-V Health Check] True
[2023-10-10T14:54:14.261] [trace] [Hyper-V Health Check] [16816] Output: True
[2023-10-10T14:54:14.261] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.261] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V | Select-Object -ExpandProperty State'
[2023-10-10T14:54:14.618] [trace] [Hyper-V Health Check] Enabled
[2023-10-10T14:54:14.618] [trace] [Hyper-V Health Check] [16816] Output: Enabled
[2023-10-10T14:54:14.618] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.618] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V-Hypervisor | Select-Object -ExpandProperty State'
[2023-10-10T14:54:14.634] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBjDCCATECBLNTUwMwCgYIKoZIzj0EAwIwUDELMAkGA1UEBhMCVVMxEjAQBgNV
BAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTllNDQtMDBmYS00NWYyLWFlYzMt
OTM5NjUzZmExOTEwMB4XDTIzMTAwNzAzMjYwOFoXDTI0MTAwNjAzMjYwOFowUDEL
MAkGA1UEBhMCVVMxEjAQBgNVBAoMCUNhbm9uaWNhbDEtMCsGA1UEAwwkNTdhMTll
NDQtMDBmYS00NWYyLWFlYzMtOTM5NjUzZmExOTEwMFkwEwYHKoZIzj0CAQYIKoZI
zj0DAQcDQgAEFfCFJ/eEpXSc0Y4rw+OSY+u+dXxojk+2n5CLQmpqGBs6jx7ha7N6
6CMpRc37QQ1k+uIeLudIA1JwCP79+RP9tzAKBggqhkjOPQQDAgNJADBGAiEA0iFi
0uX7SKLhHm0am4jayZMjXsnP6yI7zL5mtXMfkzACIQDTZDmEGhLd36wrp4QQYGBd
9bMbeH28mJR6bred8RMQ8w==
-----END CERTIFICATE-----

[2023-10-10T14:54:14.805] [trace] [Hyper-V Health Check] Enabled
[2023-10-10T14:54:14.805] [trace] [Hyper-V Health Check] [16816] Output: Enabled
[2023-10-10T14:54:14.805] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.805] [debug] [Hyper-V Health Check] [16816] Cmdlet: 'Get-Service -Name vmms | Select-Object -ExpandProperty Status'
[2023-10-10T14:54:14.821] [trace] [Hyper-V Health Check] Running
[2023-10-10T14:54:14.821] [trace] [Hyper-V Health Check] [16816] Output: Running
[2023-10-10T14:54:14.821] [debug] [Hyper-V Health Check] [16816] Cmdlet exit status is 'true'
[2023-10-10T14:54:14.833] [trace] [Hyper-V Health Check] [16816] PowerShell state changed to NotRunning
[2023-10-10T14:54:14.833] [debug] [Hyper-V Health Check] [16816] PowerShell finished successfully
[2023-10-10T14:54:14.835] [trace] [Network Listing on Windows Platform] PowerShell arguments: Get-NetAdapter, -physical, |, Select-Object, -Property, Name,MediaType,PhysicalMediaType,InterfaceDescription, |, ConvertTo-Csv, -NoTypeInformation, |, Select-Object, -Skip, 1, |, foreach, {, $_, -replace, '^"|"$|"(?=,)|(?<=,)"','', }
[2023-10-10T14:54:14.835] [trace] [Network Listing on Windows Platform] PowerShell working dir ''
[2023-10-10T14:54:14.835] [trace] [Network Listing on Windows Platform] PowerShell program 'powershell.exe'
[2023-10-10T14:54:14.835] [trace] [Network Listing on Windows Platform] [0] PowerShell state changed to Starting
[2023-10-10T14:54:14.844] [trace] [Network Listing on Windows Platform] [0] PowerShell state changed to Running
[2023-10-10T14:54:14.844] [debug] [powershell.exe] [7044] started: powershell.exe Get-NetAdapter -physical | Select-Object -Property Name,MediaType,PhysicalMediaType,InterfaceDescription | ConvertTo-Csv -NoTypeInformation | Select-Object -Skip 1 | foreach { $_ -replace '^"|"$|"(?=,)|(?<=,)"','' }
[2023-10-10T14:54:15.597] [trace] [Network Listing on Windows Platform] [7044] PowerShell state changed to NotRunning
[2023-10-10T14:54:15.597] [debug] [Network Listing on Windows Platform] [7044] PowerShell finished successfully
[2023-10-10T14:54:15.597] [trace] [Network Listing on Windows Platform] [7044] Output:
Ethernet 2,802.3,802.3,Intel(R) Ethernet Controller (3) I225-V #2
Wi-Fi,Native 802.11,Native 802.11,Intel(R) Wi-Fi 6E AX210 160MHz
[2023-10-10T14:54:15.597] [trace] [Hyper-V Switch Listing] PowerShell arguments: Get-VMSwitch, -ComputerName, localhost, |, Select-Object, -Property, Name,SwitchType,NetAdapterInterfaceDescription,Notes, |, ConvertTo-Csv, -NoTypeInformation, |, Select-Object, -Skip, 1, |, foreach, {, $_, -replace, '^"|"$|"(?=,)|(?<=,)"','', }
[2023-10-10T14:54:15.597] [trace] [Hyper-V Switch Listing] PowerShell working dir ''
[2023-10-10T14:54:15.597] [trace] [Hyper-V Switch Listing] PowerShell program 'powershell.exe'
[2023-10-10T14:54:15.597] [trace] [Hyper-V Switch Listing] [0] PowerShell state changed to Starting
[2023-10-10T14:54:15.605] [trace] [Hyper-V Switch Listing] [0] PowerShell state changed to Running
[2023-10-10T14:54:15.605] [debug] [powershell.exe] [24780] started: powershell.exe Get-VMSwitch -ComputerName localhost | Select-Object -Property Name,SwitchType,NetAdapterInterfaceDescription,Notes | ConvertTo-Csv -NoTypeInformation | Select-Object -Skip 1 | foreach { $_ -replace '^"|"$|"(?=,)|(?<=,)"','' }
[2023-10-10T14:54:16.736] [trace] [Hyper-V Switch Listing] [24780] PowerShell state changed to NotRunning
[2023-10-10T14:54:16.737] [debug] [Hyper-V Switch Listing] [24780] PowerShell finished successfully
[2023-10-10T14:54:16.737] [trace] [Hyper-V Switch Listing] [24780] Output:
Dev,Internal,,
Default Switch,Internal,,
WSL,Internal,,
Name            Type      Description
Default Switch  switch    Virtual Switch with internal networking
Dev             switch    Virtual Switch with internal networking
Ethernet 2      ethernet  Intel(R) Ethernet Controller (3) I225-V #2
WSL             switch    Virtual Switch with internal networking
Wi-Fi           wifi      Intel(R) Wi-Fi 6E AX210 160MHz
leona ~  2.994s
❯
```

Thank you for the help.

**Comment by ricab** [comment]
Great, glad you got it fixed and thanks for sharing what worked for you. Closing this now since we're tracking the core issue in #1632.

**Comment by ricab** [comment]
Duplicate of #1632

**Comment by harisraharjo** [comment]
hello @ricab sorry for pinging you. I figured that this is a similiar question so I will put a comment here. 
Is it recommended to use multipass inside wsl2? in my case I need to install homeassistant supervised in a linux environment and it's not possible with wsl2 so I have to use a vm

Thanks

**Comment by ricab** [comment]
Hi @harisraharjo. No, multipass inside WSL is not a recommended setup. Here's [how you'd normally install on Windows](https://multipass.run/docs/installing-on-windows).
